pyvo/astropy_samp/lockfile_helpers.py

```diff
@@ -0,0 +1,268 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+
+# TODO: this file should be refactored to use a more thread-safe and
+# race-condition-safe lockfile mechanism.
+
+import datetime
+import os
+import socket
+import stat
+import warnings
+from contextlib import suppress
+from urllib.parse import urlparse
+import xmlrpc.client as xmlrpc
+
+from astropy.config.paths import _find_home
```

Comment:
This uses a "hidden" function in `astropy`. Please let me know if you have a better alternative.